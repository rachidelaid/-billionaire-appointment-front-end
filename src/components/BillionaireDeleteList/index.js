import React, { useState, useEffect } from 'react';
import style from './style.module.css';

const apiURL = 'http://localhost:3000/api/billionaires'
const deleteApiURL = (id) => {
  return `http://localhost:3000/api/billionaires/${id}`
}

const getBillionaires = async () => {
  const result =
    await fetch(apiURL)
      .then((resp) => resp.json())
      .then((data) => data)
      .catch((error) => error);

  return result
}

const BillionaireDeleteList = () => {

  const [billionaires, setBillionaires] = useState([]);
  const [alert, setAlert] = useState(null);

  const setBillionairesList = async () => {
    const result = await getBillionaires();
    setBillionaires(result);
  }

  useEffect(() => {

    setBillionairesList();

  }, [])

  const deleteBillionaire = async (id) => {
    let result = {
      response: {},
      data: {}
    }

      await fetch(deleteApiURL(id), {
        method: 'DELETE'
      })
        .then((resp) => {
          result.response = resp;
          return resp.json();
        })
        .then((data) => {
          result.data = data;
          return data;
        })
        .catch((error) => error);

    return result
  }

  const handleDeleteBillionaire = async (id) => {
    const result = await deleteBillionaire(id);
    setBillionairesList();

    if (result.response.ok) {
      setAlert(result.data)
    }
  }


  const renderBillionaire = (billionaire) => {
    const b = billionaire

    return (
      <div key={b.id}className={`${style['billionaire-ctn']} ${style['d-flex']} ${style.col}`}>
        <h2>{b.name}</h2>
        <button type="button"
          onClick={() => handleDeleteBillionaire(b.id)}>DELETE</button>
      </div>
    )
  }

  const renderBillionaires = (billionaires) => {
    if (billionaires.length > 0) {
      return (<div>{
        billionaires.map((billionaire) =>
          renderBillionaire(billionaire)
        )
      }</div>
      )
    } else {
      return (
        <div>There are no billionaires!</div>
      )
    }
  }

  return (
    <div>
      <p>Hello from Delete List</p>
      {alert && 
      <div><p>{alert}</p></div>}
      <div>{renderBillionaires(billionaires)}</div>
    </div>
  )
};

export default BillionaireDeleteList;
