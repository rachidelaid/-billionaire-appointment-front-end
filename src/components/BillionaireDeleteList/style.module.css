.d-flex {
  display: flex;
}

.col {
  flex-direction: column;
}

.alert-ctn {
  color: red;
}

.click {
  cursor: pointer;
}

img {
  max-width: 200px;
}

.delete-list-component-ctn {
  align-items: center;
  justify-content: center;
}

.billionaires-ctn {
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  max-width: 100%;
  gap: 30px 50px;
}

.billionaire-ctn button {
  background-color: white;
  color: var(--green);
  text-align: center;
  font-weight: bold;
  cursor: pointer;
  border: solid 1px rgba(255, 255, 255, 0.468);
  border-radius: 15px;
  padding: 10px;
}
