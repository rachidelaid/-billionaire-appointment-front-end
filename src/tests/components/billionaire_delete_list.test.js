import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Wrapper from '../helpers/test_wrapper';
import BillionaireDeleteList from '../../components/BillionaireDeleteList';

describe('New Billionaire Page', () => {
  const user = {
    role: 'admin',
  };

  const billionaires = [{
    id: 1,
    name: 'Jeff',
    title: 'title',
    username: 'username',
    price: 949,
    image: 'url',
  }];

  const page = <Wrapper><BillionaireDeleteList user={user} billionaires={billionaires} /></Wrapper>;
  const rendered = render(page);

  it('should render the Delete Billionaires page', async () => {
    expect(rendered).toMatchSnapshot();
  });

  it('should display billionaire box', async () => {
    render(page);
    expect(screen.queryByText(/Jeff/i)).toBeInTheDocument();
    expect(screen.queryByText(/DELETE/i)).toBeInTheDocument();
  });
});
