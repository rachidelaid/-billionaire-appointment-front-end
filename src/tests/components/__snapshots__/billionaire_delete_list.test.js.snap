// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Billionaire Page should render the Add a New Billionaire page when user is admin 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col d-flex"
      >
        <form
          class="d-flex col form-"
        >
          <label
            for="name"
          >
            <p>
              Name:
            </p>
            <input
              name="name"
              type="text"
            />
          </label>
          <label
            for="title"
          >
            <p>
              Title:
            </p>
            <input
              name="title"
              type="text"
            />
          </label>
          <label
            for="price"
          >
            <p>
              Price:
            </p>
            <input
              name="price"
              step=".01"
              type="number"
            />
          </label>
          <label
            for="image"
          >
            <p>
              Image URL:
            </p>
            <input
              name="image"
              type="text"
            />
          </label>
          <label
            for="description"
          >
            <p>
              Description:
            </p>
            <textarea
              name="description"
            />
          </label>
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col d-flex"
    >
      <form
        class="d-flex col form-"
      >
        <label
          for="name"
        >
          <p>
            Name:
          </p>
          <input
            name="name"
            type="text"
          />
        </label>
        <label
          for="title"
        >
          <p>
            Title:
          </p>
          <input
            name="title"
            type="text"
          />
        </label>
        <label
          for="price"
        >
          <p>
            Price:
          </p>
          <input
            name="price"
            step=".01"
            type="number"
          />
        </label>
        <label
          for="image"
        >
          <p>
            Image URL:
          </p>
          <input
            name="image"
            type="text"
          />
        </label>
        <label
          for="description"
        >
          <p>
            Description:
          </p>
          <textarea
            name="description"
          />
        </label>
        <input
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
